File: src/main/java/com/ruthlessjailer/api/theseus/task/handler/AsyncTask.java
package com.ruthlessjailer.api.theseus.task.handler;

import lombok.Getter;

import java.util.concurrent.ScheduledFuture;

/**
 * @author RuthlessJailer
 */
@Getter
public final class AsyncTask<T> {

	public AsyncTask(final ScheduledFuture<T> task) {
		this(task, -1);
	}

	public AsyncTask(final ScheduledFuture<T> task, final int repeat) {
		this.future = task;
		this.repeat = repeat;
	}

	private final ScheduledFuture<T> future;
	private final int                repeat;

	private volatile int runs = 0;

	public synchronized void increment() { this.runs++; }
}

File: src/main/kotlin/com/ruthlessjailer/api/theseus/task/handler/AsyncTask.kt
package com.ruthlessjailer.api.theseus.task.handler

import java.util.concurrent.ScheduledFuture

/**
 * @author RuthlessJailer
 */
class AsyncTask<T>(val task: ScheduledFuture<T?>, repeat: Int = -1) : Task(repeat)